test/SlowTests/MailingList/TransformWithLoad.cs

```diff
@@ -1,234 +0,0 @@
-                    Assert.Equal(2, result.Count);
-                    Assert.NotNull(result["contacts/1"]);
-                    Assert.Null(result["contacts/2"]);
-
-                    foreach (var detail in result["contacts/1"].ContactDetails)
-                    {
-                        Assert.NotNull(detail.Id);
-                    }
-                }
-            }
-        }
-
-        [Fact]
-        public void LoadByIds()
-        {
-            using (var store = GetDocumentStore())
-            {
-                CreateData(store);
-                using (var session = store.OpenSession())
-                {
-                    var result = session.Load<ContactTransformer, ContactDto>(new[] { "contacts/1", "contacts/2" });
-                    Assert.Equal(2, result.Count);
-                    Assert.NotNull(result["contacts/1"]);
-                    Assert.Null(result["contacts/2"]);
-
-                    foreach (var detail in result["contacts/1"].ContactDetails)
-                    {
-                        Assert.NotNull(detail.Id);
-                    }
-                }
-            }
-        }
-
-        [Fact]
-        public void PlainLoadByIds()
-        {
-            using (var store = GetDocumentStore())
-            {
-                CreateData(store);
-                using (var session = store.OpenSession())
-                {
-                    var result = session.Load<Contact>(new[] {"contacts/1", "contacts/2"});
-                    Assert.Equal(2, result.Count);
-                    Assert.NotNull(result["contacts/1"]);
-                    Assert.Null(result["contacts/2"]);
-                }
-            }
-        }
-
-        private class Contact
-        {
-            public Contact()
-            {
-                DetailIds = new List<string>();
-            }
-
-            public string Id { get; set; }
-            public string Name { get; set; }
-            public List<string> DetailIds { get; set; }
-        }
-
-        private class ContactDto
-        {
-            public string ContactId { get; set; }
-            public string ContactName { get; set; }
-            public List<Detail> ContactDetails { get; set; }
-            public string MetaVal { get; set; }
-        }
-
-        private class ContactTransformer : AbstractTransformerCreationTask<Contact>
-        {
-            public ContactTransformer()
-            {
-                TransformResults = contacts => from c in contacts
-                                               select new
-                                               {
-                                                   ContactId = c.Id,
-                                                   ContactName = c.Name,
-                                                   ContactDetails = LoadDocument<Detail>(c.DetailIds),
-                                                   MetaVal = MetadataFor(c).Value<string>("Val")
```

Comment:
Restore candidate: create a test to make sure function can access metadata from a doc